Automattic/sensei
# Fix editor undo for auto-inserted blocks & quiz block

These mostly make the undo experience tolerable for the lesson & course editors. The post is still marked as dirty after saving most of the time, regardless of the quiz block. Metaboxes and various editor internals are probably at fault. But the whole saved/unsaved thing doesn't seem to be too prominent in the editor, it's not even indicated if the post is already published.

### Changes proposed in this Pull Request

* Use `__unstableMarkNextChangeAsNotPersistent` action if available to mark automatic draft block insertion as automatic
* Don't remove auto-inserted block on focus loss, just hide it with CSS
* Only update `_quiz_has_questions` meta if it should change
* Also handle a case where setting the above meta to null a second time would throw errors
* Toggle the `View Quiz` button in the editor based on the quiz block state

### Testing instructions

* Add a quiz block and edit various questions & other things in the lesson. Try undo-ing, and redo-ing everything.
* Try adding/removing the quiz block or all of its questions and check that the lesson has/doesn't have quiz. (`View Quiz` button showing up in the frontend)
* Check that `View Quiz` button in the editor also shows up depending on the having a quiz block with questions
* Check that auto-inserted lesson blocks in the course outline block modules also work


### Screenshot / Video
